src/core/stream.js

```diff
@@ -942,7 +942,16 @@ var JpegStream = (function JpegStreamClosure() {
   JpegStream.prototype.isNativelySupported =
       function JpegStream_isNativelySupported(xref, res) {
     var cs = ColorSpace.parse(this.dict.get('ColorSpace', 'CS'), xref, res);
-    return cs.name === 'DeviceGray' || cs.name === 'DeviceRGB';
+    if (cs.name !== 'DeviceGray' && cs.name !== 'DeviceRGB') {
+      return false;
+    }
+    if (this.dict) {
+      var decode = this.dict.get('Decode', 'D');
+      if (decode && isArray(decode) && !cs.isDefaultDecode(decode)) {
+        return false;
+      }
+    }
+    return true;
```

Comment:
- The `if (this.dict)` check is too defensive. At the first line, `this.dict` is already assumed to be non-void.
- `decode && isArray(decode)` can be simplified to `isArray(decode)`, because `isArray` is certainly not going to return true for falsey objects.

I suggest to put the `isArray` check in `isDefaultDecode` (if needed), and replace the code with

```
return (cs.name === 'DeviceGray' || cs.name === 'DeviceRGB') &&
  cs.isDefaultDecode(this.dict.get('Decode', 'd'));
```